collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -253,6 +255,7 @@ protected void onCleared() {
     }
 
     private void updateIndex() {
+        Timber.w("Update formIndex");
```

Comment:
I added this to check if the value in livedata is set by us or maybe it's a restored state as I described in my comment https://github.com/getodk/collect/issues/5241#issuecomment-1206333890